modules/game/src/main/BSONHandlers.scala

```diff
@@ -94,7 +96,15 @@ object BSONHandlers {
         castleLastMoveTime = r.get[CastleLastMoveTime](castleLastMoveTime)(CastleLastMoveTime.castleLastMoveTimeBSONHandler),
         unmovedRooks = r.getO[UnmovedRooks](unmovedRooks) | UnmovedRooks.default,
         daysPerTurn = r intO daysPerTurn,
-        clockHistory = ClockHistory(r bytesO moveTimes),
+        binaryMoveTimes = r bytesO moveTimes,
+        clockHistory = for {
+          clk <- gameClock
+          start = clk.limit.seconds
+          ew = (clk.remainingTime(White) * 1000).toLong.millis
+          eb = (clk.remainingTime(Black) * 1000).toLong.millis
+          bw <- r bytesO whiteClockHistory
+          bb <- r bytesO blackClockHistory
+        } yield BinaryFormat.clockHistory.read(start, ew, eb, bw, bb, r intD startedAtTurn, nbTurns),
```

Comment:
Doesn't this make the clock history decoding eager, instead of lazy?

With the former `binaryMoveTimes = r bytesO moveTimes`, the `ByteArray` was read immediately, but it was only converted to actual move times on demand. I would guess that most instances of the Game object don't care about move times, so it makes sense to delay their computation if possible.